osu.Game/Screens/Play/Player.cs

```diff
@@ -816,6 +798,51 @@ private void progressToResults(bool withDelay)
             Scheduler.Add(resultsDisplayDelegate);
         }
 
+        /// <summary>
+        /// Asynchronously run score preparation operations (database import, online submission etc.).
+        /// </summary>
+        /// <param name="forceImport">Whether the score should be imported even if non-passing (or the current configuration doesn't allow for it).</param>
+        /// <returns>The final score.</returns>
+        [ItemCanBeNull]
+        private Task<ScoreInfo> prepareAndImportScoreAsync(bool forceImport = false)
+        {
+            // Ensure we are not writing to the replay any more, as we are about to consume and store the score.
+            DrawableRuleset.SetRecordTarget(null);
```

Comment:
Should I be worried that this is changing `Drawable` states via calling `.Expire()` internally, yet it runs on a TPL thread when the button's clicked from the results screen?

![image](https://user-images.githubusercontent.com/1329837/223923959-61ccc490-0bf2-4f82-965e-9ce4f058e69f.png)

Reply:
It's pretty dodge when you look at it that way, but the `null` path at most [calls `Expire()` on a drawable](https://github.com/ppy/osu/blob/7bc8908ca9c026fed1d831eb6e58df7624a8d614/osu.Game/Rulesets/UI/RulesetInputManager.cs#L45) so i guess it's mostly safe...